src/service.rs

```diff
@@ -37,21 +41,18 @@ fn start_http_server(cfg: &Config, networks: Arc<Networks>) -> iron::Listening {
 }
 
 // TODO: make this a struct which receives a shutdown message on a channel and then wraps itself up
-fn start_networks_refreshers(cfg: Config) -> Vec<thread::JoinHandle<()>> {
+fn start_networks_refreshers(networks: Arc<Networks>) -> Vec<thread::JoinHandle<()>> {
     let mut threads = vec![];
-    match cfg.get_networks() {
-        Err(err) => panic!("Unable to get networks: {:?}", err),
-        Ok(networks) => {
-            for (label, mut net) in networks.into_iter() {
-                threads.push(thread::spawn(move || {
-                    loop {
-                        refresh_network(&label, &mut net);
-                        // In case of an error, wait a while before retrying.
-                        thread::sleep(Duration::from_secs(10));
-                    }
-                }));
+    for (label, net) in networks.iter() {
+        let label = label.to_owned();
+        let mut net = net.clone();
```

Comment:
So when you clone a `Network` here, the storage instance inside it will not be actually cloned because it's in another `Arc`, right?

Reply:
Yes. It just copies the Arc (ie increments the refcount). I chose to copy the entire network thing instead of trying to lock the entire structure since the other fields we are only reading and not mutating (ever).

As for testing, it definitely shares the storage. I tested it during initial sync and after it's synced and doing the occasional re-syncs. For mainnet I tried querying for a block and initially receive doesn't exist, then later after it syncs far enough to receive it, it will return that block, before it's fully done syncing. There's usually very little delay between responses, but sometimes it can be a couple seconds during some operations.